Allow installation as a package with setup.py

vtantia opened the pull request:
This contains changes to -
- Create setup.py, update requirements.txt and installation instructions in README.md
- Changes in gossip/ad_psgd.py to allow creation of a package

vtantia committed:
Allow installation as a package with setup.py

mikerabbat commented on setup.py:
```diff
@@ -0,0 +1,47 @@
+        license="Attribution-NonCommercial 4.0 International",
+        python_requires=">=3.6",
+        packages=find_packages(),
+        # package_dir={"": "gossip"},
```
Delete this line rather than commenting out?

vtantia replied:
Thanks done!

mikerabbat commented on requirements.txt:
```diff
@@ -1,4 +1,3 @@
-https://download.pytorch.org/whl/cu90/torch-1.0.0-cp36-cp36m-linux_x86_64.whl
+torch==1.0.0
```
Does this need to be == or would >= be ok?

vtantia replied:
With the current version of the open source code, this needs to be ==. We need to make a few changes to allow 1.4.0. Do you think doing that later on in a separate PR would be okay?

mikerabbat replied:
Yes, ok to do this later, thanks! Please create an issue describing what needs to be changed to make the code compatible with PyTorch 1.4.0

vtantia replied:
Done thanks!

mikerabbat reviewed (changes requested):
In addition to the few small other comments, it seems non-ideal to have two versions of get_tcp_interface_name in the repo, one in experiment_utils and another in gossip/utils. Can we refactor to keep only one? It would also be good to be consistent across gossip_sgd.py and gossip_sgd_adpsgd.py in how/where this check gets performed. Is the main script responsible for getting the interface, or the *GossipDataParallel module? Seems like functionality that is better left for the calling script for now, in case the user wants to specify a particular interface.

vtantia committed:
Delete commented out line

vtantia committed:
Refactor to have single copy of get_tcp_interface_name

vtantia committed:
Fix error in importing pandas

vtantia commented:
Thanks for the comments! I have addressed the inline comments and refactored the code to have one version of get_tcp_interface_name. Please let me know what you think!

mikerabbat reviewed (approved):
Looks good to me, thanks!

As a follow-up, please open an issue summarizing what changes need to be made to be compatible with PyTorch 1.4.0.
